listenbrainz_spark/recommendations/candidate_sets.py

```diff
@@ -24,211 +24,217 @@
 # Candidate Set HTML is generated if set to true.
 SAVE_CANDIDATE_HTML = True
 
-def get_listens_for_rec_generation_window(mapped_df):
-    """ Get listens to fetch top artists.
+# Some useful dataframe fields/columns.
+# top_artists_df:
+#   [
+#       'mb_artist_credit_id', 'msb_artist_credit_name_matchable', 'user_name'
+#   ]
+#
+# top_artists_candidate_set_df:
+#   [
+#       'user_id', 'recording_id'
+#   ]
+#
+# top_similar_artists_df:
+#   [
+#       'top_artist_credit_id', 'top_artist_name', 'similar_artist_credit_id', 'similar_artist_name'
+#       'score', 'user_name', 'rank'
+#   ]
+#
+# top_similar_artists_candidate_set_df:
+#   [
+#       'user_id', 'recording_id'
+#   ]
+
+
+def get_listens_to_fetch_top_artists(mapped_df):
+    """ Get listens of past X days to fetch top artists where X = RECOMMENDATION_GENERATION_WINDOW.
 
         Args:
-            mapped_df (dataframe): Dataframe with all the columns/fields that a typical listen has.
+            mapped_df (dataframe): listens mapped with msid_mbid_mapping. Refer to candidate_sets.py
+                                   for dataframe columns.
+
+        Returns:
+            mapped_listens_subset (dataframe): A subset of mapped_df containing user history.
     """
-    df = mapped_df.select('*') \
-        .where((col('listened_at') >= to_timestamp(date_sub(current_timestamp(),
-        config.RECOMMENDATION_GENERATION_WINDOW))) & (col('listened_at') <= current_timestamp()))
-    return df
+    mapped_listens_subset = mapped_df.select('*') \
+                                     .where((col('listened_at') >= to_timestamp(date_sub(current_timestamp(),
+                                             config.RECOMMENDATION_GENERATION_WINDOW))) &
+                                            (col('listened_at') <= current_timestamp()))
+    return mapped_listens_subset
+
 
-def get_top_artists(df, user_name):
-    """ Get top artists listened to by the user.
+def get_top_artists(mapped_listens_subset):
+    """ Get top artists listened to by users who have a listening history in
+        the past X days where X = RECOMMENDATION_GENERATION_WINDOW.
 
         Args:
-            df (dataframe): Dataframe to containing user history.
-            user_name (str): Name of the user.
+            df (dataframe): A subset of mapped_df containing user history.
```

Comment:
```suggestion
            mapped_listens_subset (dataframe): A subset of mapped_df containing user history.
```

Reply:
Thanks. I have already made these changes in a separate commit. Will puch it soon :)